sdk/servicebus/service-bus/samples-dev/topicSubscriptionWithRuleOperationsSample.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT Licence.
+
+/**
+ * This sample demonstrates the topic filter functionality in Service Bus.
+ * It follows similar functionality to TopicSubscriptionWithRuleOperationsSample/program.cs
+ *
+ *
+ * @summary Demonstrates how to filter messages in Service Bus
+ * @azsdk-weight 100
+ */
+
+import { ServiceBusClient, ServiceBusAdministrationClient, ServiceBusMessage } from "@azure/service-bus";
+import { DEFAULT_RULE_NAME } from "../src/util/constants";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Define connection string and related Service Bus entity names here
+const connectionString = process.env.SERVICEBUS_CONNECTION_STRING || "<connection string>";
+const topicName = "TopicSubscriptionWithRuleOperationsSample";
+
+const firstSetOfMessages: ServiceBusMessage[] = [
+  { subject: "Red", body: "test"},
+  { subject: "Red", body: "test", correlationId: "notimportant"},
```

Comment:
[NIT]

Is this a good use of the correlationId?
I imagined it to be a unique identifier and not a mere label.

Reply:
I'm not too sure - my point of reference is currently the lines here

https://github.com/Azure/azure-sdk-for-net/blob/db4973e3cbc69644ba5d98d82748a483c1432946/sdk/servicebus/Azure.Messaging.ServiceBus/samples/TopicFilters/Program.cs#L151-L161

I'll see if this works and I can definitely revise after the fact.